pkg/controllers/ingressclass/ingressclass.go

```diff
@@ -283,25 +279,36 @@ func (c *Controller) ensureLoadBalancer(ic *networkingv1.IngressClass) error {
 
 	if *lb.Id != util.GetIngressClassLoadBalancerId(ic) {
 		klog.InfoS("Adding load balancer id to ingress class", "lbId", *lb.Id, "ingressClass", klog.KObj(ic))
-
-		patchBytes := []byte(fmt.Sprintf(`{"metadata":{"annotations":{"%s":"%s"}}}`, util.IngressClassLoadBalancerIdAnnotation, *lb.Id))
-
-		err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-			_, err := c.client.NetworkingV1().IngressClasses().Patch(context.TODO(), ic.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
-			return err
-		})
-
-		if apierrors.IsConflict(err) {
-			return errors.Wrapf(err, "updateMaxRetries(%d) limit was reached while attempting to add load balancer id annotation", retry.DefaultBackoff.Steps)
+		patchError, done := util.PatchIngressClassWithAnnotation(c.client, ic, util.IngressClassLoadBalancerIdAnnotation, *lb.Id)
+		if done {
+			return patchError
 		}
+	}
 
+	// Add Web Application Firewall to LB
+	if c.wafClient != nil {
+		err = c.setupWebApplicationFirewall(ic, compartmentId, lb.Id)
 		if err != nil {
 			return err
 		}
 	}
 
 	klog.V(4).InfoS("checking if updates are required for load balancer", "ingressClass", klog.KObj(ic))
+	return nil
+}
 
+func (c *Controller) setupWebApplicationFirewall(ic *networkingv1.IngressClass, compartmentId *string, lbId *string) error {
+	firewall, err, err2, done := c.wafClient.GetFireWallId(c.client, ic, compartmentId, lbId)
```

Comment:
Can we add a comment on what would be value of done here and on what case we return err2? (or update variables accordingly?)

Reply:
Done